proto/skaffold.proto

```diff
@@ -231,4 +262,35 @@ enum ErrorCode {
 //    COULD_NOT_GET_DOCKER_CLIENT = 7;
 //    COULD_NOT_GET_LOCAL_CLUSTER = 8;
 //    BUILDER_CLEANUP = 9;
+}
+
+enum BuilderType {
+    UNKNOWN_BUILDER_TYPE = 0;
+    JIB = 1;
+    BAZEL = 2;
+    BUILDPACKS = 3;
+    CUSTOM = 4;
+    KANIKO = 5;
+    DOCKER = 6;
+}
+
+enum BuildType {
+    UNKNOWN_BUILD_TYPE =0 ;
+    CLUSTER = 1;
+    GCB = 2;
+    LOCAL = 3;
+}
+
+enum DeployerType {
+    UNKNOWN_DEPLOYER_TYPE =0 ;
+    HELM = 1;
+    KUSTOMIZE = 2;
+    KUBECTL = 3;
+}
+
+enum ClusterType {
```

Comment:
seems like we might want to list out some of the other common k8s cluster providers here. otherwise users are going to see "Cluster Type: Other" which doesn't seem that useful?

Reply:
so for Privacy concerns, we can't make a distinction which "Other" or competitor cloud provider service users are using :)

Reply:
Can we differentiate with other opensource distributions?  (k3d, microk8s, ...?)